pkg/services/alerting/extractor.go

```diff
@@ -203,16 +226,49 @@ func (e *DashAlertExtractor) getAlertFromPanels(jsonWithPanels *simplejson.Json,
 	return alerts, nil
 }
 
-func validateAlertRule(alert *models.Alert) bool {
-	return alert.ValidToSave()
+func validateAlertRule(alert *models.Alert) (ok bool, reason string) {
+	ok = alert.ValidToSave()
+	if !ok {
+		reason = fmt.Sprintf("Panel id is not correct, alertName=%v, panelId=%v", alert.Name, alert.PanelId)
+	}
+	return ok, reason
+}
+
+func validAlertJSON(alert *models.Alert) (ok bool, reason string) {
+	warnings := strings.Builder{}
+	for _, v := range alert.Settings.Get("notifications").MustArray() {
+		jsonModel := simplejson.NewFromAny(v)
+		if id, err := jsonModel.Get("id").Int64(); err == nil {
+			_, err := translateNotificationIDToUID(id, alert.OrgId)
+			if err != nil {
+				ok = false
+				if warnings.Len() > 0 {
+					warnings.WriteString("\n")
+				}
+				warnings.WriteString(fmt.Sprintf("Alert contains notification identified by incorrect id, alertName=%v, panelId=%v, notificationId=%v", alert.Name, alert.PanelId, id))
+			}
+		}
```

Comment:
It's best to avoid nesting:
```suggestion
		id, err := jsonModel.Get("id").Int64()
		if err != nil {
			continue
		}
		
		if _, err := translateNotificationIDToUID(id, alert.OrgId); err == nil {
			continue
		}

		ok = false
		if warnings.Len() > 0 {
			warnings.WriteString("\n")
		}
		warnings.WriteString(fmt.Sprintf("Alert contains notification identified by incorrect id, alertName=%v, panelId=%v, notificationId=%v", alert.Name, alert.PanelId, id))	
```